lua/core/autocmds.lua

```diff
@@ -11,6 +11,9 @@ vim.cmd [[ au TermOpen term://* setlocal nonumber norelativenumber | setfiletype
 -- Don't show status line on certain windows
 vim.cmd [[ autocmd BufEnter,BufWinEnter,FileType,WinEnter * lua require("core.utils").hide_statusline() ]]
 
+--auto close file exploer when quiting incase a single buffer is left
+vim.cmd([[ autocmd BufEnter * if (winnr("$") == 1 && &filetype == 'nvimtree') | q | endif ]])
```

Comment:
I think this is good,
my only suggestion is that we may wish to add a `default_config` option to enable/disable this behavior.

What do you think?